frontend/src/store/objectStore.ts

```diff
@@ -158,9 +158,13 @@ export const useObjectStore = defineStore('object', () => {
 
           // Merge and assign
           state.objects.value = difference.concat(response);
-          // Track all the object IDs that the user would have access to in the table (even if filters are applied)
-          if(!tagset && !metadata) {
-            state.unfilteredObjectIds.value = state.objects.value.map((o) => o.id);
+          
+          // Track all the object IDs in this bucket that the user would have access to in the table 
+          // (even if filters are applied)
+          if(!tagset?.length && !metadata?.length) {
+            state.unfilteredObjectIds.value = state.objects.value
+              .filter( (x) => x.bucketId === params.bucketId )
+              .map((o) => o.id);
```

Comment:
This works for now, but if and when we decide to revisit the idea of aggregating all objects across all buckets a user has access to, this new logic will be in conflict with this. Something we'll have to assess down the line if we decide to implement a cross-bucket view like that.

Reply:
Yeah this will need to be adjusted if it's across buckets when/if that becomes a feature